openpype/pipeline/file_collection.py

```diff
@@ -0,0 +1,274 @@
+"""Utility functions for file sequences and collections
+"""
+import os
+import re
+import clique
+from openpype.lib import Logger
+from openpype.pipeline import frames
+
+
+def collect_filepaths_from_sequential_path(
+    file_path,
+    frame_start=None,
+    frame_end=None,
+    only_existing=False,
+):
+    """Generate expected files from path
+
+    Args:
+        file_path (str): Path to generate expected files from
+        frame_start (Optional[int]): Start frame of the sequence
+        frame_end (Optional[int]): End frame of the sequence
+        only_existing (Optional[bool]): Ensure that files exists.
+
+    Returns:
+        Any[list[str], str]: List of absolute paths to files
+            (frames of a sequence) or path if single file
+    """
+
+    dirpath = os.path.dirname(file_path)
+    filename = os.path.basename(file_path)
+
+    formattable_string = convert_filename_to_formattable_string(
+        filename)
+
+    if (
+        not formattable_string
+        or frame_start is None
+        or frame_end is None
+    ):
+        return file_path
+
+    expected_files = []
+    for frame in range(int(frame_start), (int(frame_end) + 1)):
+        frame_file_path = os.path.join(
+            dirpath, formattable_string.format(frame)
+        )
+        # normalize path
+        frame_file_path = os.path.normpath(frame_file_path)
```

Comment:
We don't need to do this per file right, we can do outside the loop
```python
dirpath = os.path.normpath(dirpath)
```
And `os.path.join` will always join in the normpath way, right?